### PR TITLE
py-transonic: new package

### PR DESCRIPTION
**Pull request by paugier**
This PR adds Transonic https://pypi.org/project/transonic.

It works as expected for other Spack packages I'm working on (https://foss.heptapod.net/fluiddyn/fluiddyn/-/merge_requests/128).

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-transonic/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    version("0.7.2", sha256="d0c39c13b535df4f121a8a378efc42e3d3bf4e49536d131e6d26e9fe7d5a5bf4")
+    version("0.7.1", sha256="dcc59f1936d09129c800629cd4e6812571a74afe40dadd8193940b545e6ef03e")
+
+    depends_on("python@3.9:", type=("build", "run"))
+    depends_on("py-pdm-backend", type="build")
+
+    with default_args(type="run"):
+        depends_on("py-numpy")
+        depends_on("py-beniget")
+        depends_on("py-gast")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-transonic/package.py]:
```suggestion
        depends_on("py-beniget@0.4")
        depends_on("py-gast@0.5")
```

**Review by wdconinc — COMMENTED** [review]
Thanks for the new package.

Checksums 0.7.1 - 0.7.2 verified.

Some comments on dependencies.

File: var/spack/repos/builtin/packages/py-transonic/package.py (hunk 1 above)
Comment (paugier) [reply var/spack/repos/builtin/packages/py-transonic/package.py]: Thanks for looking at this! I added these changes and rebased.

**Review by wdconinc — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**